<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>636</width>
    <height>465</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QLabel" name="label_direction">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>40</y>
     <width>61</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>direction</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_lport">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>90</y>
     <width>71</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>local port</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_raddr">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>140</y>
     <width>111</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>remote address</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_raddr_2">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>190</y>
     <width>111</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>remote port</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_raddr_3">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>240</y>
     <width>111</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>protocol</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_raddr_4">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>280</y>
     <width>111</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>target</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="lport">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>80</y>
     <width>191</width>
     <height>32</height>
    </rect>
   </property>
  </widget>
  <widget class="QLineEdit" name="raddr">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>130</y>
     <width>191</width>
     <height>32</height>
    </rect>
   </property>
  </widget>
  <widget class="QLineEdit" name="rport">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>180</y>
     <width>191</width>
     <height>32</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_confirm">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>420</y>
     <width>93</width>
     <height>34</height>
    </rect>
   </property>
   <property name="text">
    <string>confirm</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_cancel">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>420</y>
     <width>93</width>
     <height>34</height>
    </rect>
   </property>
   <property name="text">
    <string>cancel</string>
   </property>
  </widget>
  <widget class="QWidget" name="widget_direction" native="true">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>10</y>
     <width>421</width>
     <height>61</height>
    </rect>
   </property>
   <widget class="QRadioButton" name="direction_in">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>30</y>
      <width>112</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>IN</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="direction_out">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>112</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>OUT</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <zorder>direction_in</zorder>
   <zorder>direction_in</zorder>
   <zorder>direction_out</zorder>
  </widget>
  <widget class="QWidget" name="widget_protocol" native="true">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>210</y>
     <width>291</width>
     <height>51</height>
    </rect>
   </property>
   <widget class="QRadioButton" name="protocol_tcp">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>112</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>&amp;tcp</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="protocol_udp">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>30</y>
      <width>112</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>&amp;udp</string>
    </property>
   </widget>
  </widget>
  <widget class="QWidget" name="widget_target" native="true">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>270</y>
     <width>291</width>
     <height>51</height>
    </rect>
   </property>
   <widget class="QRadioButton" name="target_drop">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>112</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>drop</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="target_accept">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>10</y>
      <width>112</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>accept</string>
    </property>
   </widget>
  </widget>
  <widget class="QPlainTextEdit" name="plainTextEdit">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>310</y>
     <width>561</width>
     <height>101</height>
    </rect>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
   <property name="overwriteMode">
    <bool>false</bool>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
